app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt

```diff
@@ -238,15 +238,39 @@ class DashboardFragment : BaseFragment<FragmentDashboardPlusBinding, DashboardVM
         }
     }
 
-    private fun updateLastUpdateDateAndTime(){
-        if (viewModel.lastUpdateTime.value != null && viewModel.lastUpdateDate.value != null) {
-            dash_card_no_risky_encounter.card_subtitle = resources.getString(
-                R.string.dashboard_body_no_contact,
-                viewModel.lastUpdateDate.value,
-                viewModel.lastUpdateTime.value
-            )
+    private fun updateLastUpdateDateAndTime() {
+        if (viewModel.lastExposureDate.value == null) {
+            if (viewModel.lastUpdateTime.value != null && viewModel.lastUpdateDate.value != null) {
+                dash_card_no_risky_encounter.card_subtitle = "${
+                    resources.getString(
+                        R.string.dashboard_body_no_contact,
+                        viewModel.lastUpdateDate.value,
+                        viewModel.lastUpdateTime.value
+                    )
+                }\n${AppConfig.encounterUpdateFrequency}"
+            } else {
+                dash_card_no_risky_encounter.card_subtitle = ""
+            }
         } else {
-            dash_card_no_risky_encounter.card_subtitle = ""
+            if (viewModel.lastUpdateTime.value != null && viewModel.lastUpdateDate.value != null) {
+                dash_card_risky_encounter.card_subtitle = "${
+                    resources.getString(
+                        R.string.dashboard_risky_encounter_subtitle_bad,
+                        viewModel.lastExposureDate.value
+                    )
+                }\n\n${
+                    resources.getString(
```

Comment:
Tohle je code duplication, chtělo by to dát do proměnné. A obecně dělat takhle složitou logiku uvnitř stringu je nepřehledné, mělo by to být v proměnné.